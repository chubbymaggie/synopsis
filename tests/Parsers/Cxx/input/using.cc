namespace Foo
{
  int x;
}

void func()
{
  using namespace Foo;
  x;
}

void func2()
{
  using namespace Foo = Bar;
  Bar::x;
}

void func3()
{
  Foo::x;
}

void func4()
{
  using Foo::x;
  x;
}
