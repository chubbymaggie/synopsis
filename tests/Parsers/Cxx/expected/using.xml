<?xml version='1.0' encoding='ISO-8859-1'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Module" file="Parsers/Cxx/input/using.cc" language="C++" line="0" name="Foo" type="namespace">
   <declarations>
    <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/using.cc" language="C++" line="3" name="Foo.x" type="namespace variable">
     <vtype>
      <instance class="Synopsis.Type.Base" language="C++" name="int"/>
     </vtype>
    </instance>
    <instance accessibility="0" class="Synopsis.AST.Builtin" file="Parsers/Cxx/input/using.cc" language="C++" line="3" name="EOS" type="EOS"/>
   </declarations>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using.cc" language="C++" line="6" name="func()" realname="func" type="function">
   <returnType>
    <instance class="Synopsis.Type.Base" language="C++" name="void"/>
   </returnType>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using.cc" language="C++" line="12" name="func2()" realname="func2" type="function">
   <returnType/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using.cc" language="C++" line="18" name="func3()" realname="func3" type="function">
   <returnType/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using.cc" language="C++" line="23" name="func4()" realname="func4" type="function">
   <returnType/>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="EOS">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.x">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="func()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="func2()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="func3()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="func4()">
   <declaration/>
  </instance>
 </types>
 <files>
  <file>
   Parsers/Cxx/input/using.cc
  </file>
 </files>
</ast>
